common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/adaptors/ProtoMqttAdaptor.java

```diff
@@ -45,20 +50,24 @@
 
     @Override
     public TransportProtos.PostTelemetryMsg convertToPostTelemetry(MqttDeviceAwareSessionContext ctx, MqttPublishMessage inbound) throws AdaptorException {
+        DeviceSessionCtx deviceSessionCtx = (DeviceSessionCtx) ctx;
         byte[] bytes = toBytes(inbound.payload());
+        Descriptors.Descriptor telemetryDynamicMsgDescriptor = getDescriptor(deviceSessionCtx.getTelemetryDynamicMsgDescriptor());
         try {
-            return ProtoConverter.convertToTelemetryProto(bytes);
-        } catch (InvalidProtocolBufferException | IllegalArgumentException e) {
+            return JsonConverter.convertToTelemetryProto(new JsonParser().parse(dynamicMsgToJson(bytes, telemetryDynamicMsgDescriptor)));
```

Comment:
Please add the TODO to improve this in the future releases. We need two types of improvements:
1) No need to convert to JSON and back. If we know the structure of the message we can dynamically create our PostTelemetryMsg without JSON
2) Need to support Protobuf for all other methods (and publish the schema somewhere)